### ISSUE TITLE
Set custom parser via CLIEngine options

### ISSUE THREAD

**Issue by feross** [eslint/eslint#3127]
Using latest ESLint, there's not an easy way to set a custom parser via the node.js API.

I expect this to work:

```js
var CLIEngine = require("eslint").CLIEngine;
var cli = new CLIEngine({
    rules: {
        semi: 2
    },
    parser: 'babel-eslint'
});
var report = cli.executeOnFiles(["myfile.js", "lib/"]);
```

Instead I have to write the config file to disk and use the `configFile` option in order to get `CLIEngine` to recognize the `parser` option.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
Per the [docs](http://eslint.org/docs/developer-guide/nodejs-api#cliengine), this isn't a supported option. Are you requesting an enhancement?

**Comment by feross** [comment]
Yep, I'm requesting an enhancement. I find it odd that `CLIEngine` and `.eslintrc` are almost identical, but have a few differences like missing the `parser` option, for instance.

**Comment by nzakas** [comment]
`CLIEngine` reflects the options available on the command line, some of which are represented in `.eslintrc` and some of which are not. The mapping is really between the command line options and `.eslintrc`, and `CLIEngine` just happens to be in the middle.

In any event, this seems like a good option to have on the command line, so we'd need to add a `--parser` flag as well as an option for `CLIEngine`.

**Comment by vinsonchuong** [comment]
@feross, I'm currently using:

```js
new CLIEngine({
  '...',
  baseConfig: {
    parser: 'babel-eslint',
    '...'
  },
  '...'
})
```

I imagine that using `baseConfig` has the benefit of allowing per-directory overrides via `.eslintrc`. So I probably prefer to even have `rules` inside of `baseConfig`

**Comment by feross** [comment]
@vinsonchuong thanks for sharing!

**Comment by nzakas** [comment]
`baseConfig` is the lowest level of priority wheras the arguments passed to the CLI are highest, so not exactly the same.
